### PR TITLE
atan2 -> atan

### LINKED ISSUE #27248: `atan2` as 2-arg method of `atan`
Should `atan2` become a 2-argument method of `atan`?

The main argument against would be convention: `atan2` is well-established, going back at least to Fortran IV (1961).

The only languages I have found that do combine them are [Mathematica `ArcTan`](http://reference.wolfram.com/language/ref/ArcTan.html) and [Common Lisp `atan`](http://clhs.lisp.se/Body/f_asin_.htm).

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #27248.

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: base/math.jl [hunk 1]
```diff
@@ -231,11 +231,13 @@ Compute hyperbolic tangent of `x`.
...
 """
-    atan(x)
+    atan(y)
+    atan(y, x)
 
-Compute the inverse tangent of `x`, where the output is in radians.
+Compute the inverse tangent of `y`, where the output is in radians.
 """
 atan(x::Number)
+atan(x::Number)
```
Comment (ararslan) [inline base/math.jl]: Duplicate of the line above it?

**Comment by andyferris** [comment]
I like it!

**Comment by ViralBShah** [comment]
Rebased on master.

**Comment by ViralBShah** [comment]
Having two ways to do this doesn't seem ideal, given that everywhere else, we have strictly gone ahead with doing things the right way. Perhaps removing `atan2` and just having the new behaviour is best in the long run.

**Comment by ViralBShah** [comment]
Wouldn't people naturally look in the `atan` documentation? Maybe just have a doc entry for `atan2`? Otherwise, why do something special only for this?

**Review by Sacha0 — APPROVED** [review]
LGTM modulo CI! Thanks Simon! :)

**Comment by StefanKarpinski** [comment]
We could have a permanent deprecation for `atan2` to `atan` kind of like we do for `**` to help people coming from Python.

**Pull request merged**